vignettes/GetStarted.Rmd

```diff
@@ -0,0 +1,100 @@
+---
+title: "Get started with the bayesian package"
+author: "Paul-Christian Bürkner & Hamada S. Badr"
+date: "`r Sys.Date()`"
+output:
+  rmarkdown::html_vignette:
+    keep_md: true
+vignette: >
+  %\VignetteIndexEntry{Getting Started}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r}
+library(recipes)
+library(workflows)
+library(bayesian)
+```
+
+As a simple example, we will model the seizure counts in epileptic patients to
+investigate whether the treatment (represented by variable `Trt`) can reduce the
+seizure counts and whether the effect of the treatment varies with the baseline
+number of seizures a person had before treatment (variable `Base`) and with the
+age of the person (variable `Age)`. As we have multiple observations per `person`,
+a group-level intercept is incorporated to account for the resulting dependency
+in the data. In a first step, we use the `recipes` package to prepare (a recipe
+for) the `epilepsy` data. This data set is shipped with the `brms` package,
+which is automatically loaded by `bayesian`.
+
+```{r}
+epi_recipe <- epilepsy %>% 
+  recipe() %>%
+  update_role(count, new_role = "outcome") %>%
+  update_role(Trt, Age, Base, patient, new_role = "predictor") %>%
+  update_role(patient) %>%
```

Comment:
The default `role` is `predictor`; so, let's add explicit roles and may add a role as `group`.
```suggestion
  update_role(patient, new_role = "outcome") %>%
  add_role(patient, new_role = "group") %>%
```

Reply:
I tried role `group` and it failed later on when trying to fit the model. Does it work for you?

Reply:
You need to add a "group" role after setting it as a predictor (to make it available in the processed data passed to the model fit, `bayesian_fit`). Any new roles should be ok.

Reply:
I see. Let me try.

Reply:
Nope, still fails for me.

Reply:
Ah wait. I need to `add` the role. Sorry.

Reply:
Yes, `update_role` sets a role while `add_role` add another one. It's a good practice to show how to identify the roles in the recipe, even if it isn't used, for reference. But, we could also select variable automatically using the added role. For example, apply a transformation step to all predictors and select or exclude `has_role("group")`.

Reply:
I have never worked with recipes myself and don't know about recommended specifications. Feel free to change it as you see fit.